### ISSUE TITLE
How the PiWeb API writes the output of measurements of  calculation characteristic

### ISSUE THREAD

**Issue by zdh5186** [ZEISS-PiWeb/PiWeb-Api#75]
**Used version**
PiWeb 7.4.5

**Question**
Your detailed question
- PiWeb API can measurements of  normal characteristic export , but measurements of  calculation characteristic can not export.
- I want to know how to write code that export measurements of  calculation characteristic

**Comment by czthiele** [comment]
Hello @zdh5186,

calculated characteristics are only evaluated on clients, e.g. Planner or Monitor using the calculation rule of the characteristic. They are calculated in realtime and their values are **not** stored in the database, this is why you currently have no possibility to export them directly using the API.

**Comment by zdh5186** [comment]
> Hello @zdh5186,
> 
> calculated characteristics are only evaluated on clients, e.g. Planner or Monitor using the calculation rule of the characteristic. They are calculated in realtime and their values are **not** stored in the database, this is why you currently have no possibility to export them directly using the API.

That is, the customer needs to write the calculation code again through the API to be able to generate the measurements of calculated characteristics , right ?

**Comment by zdh5186** [comment]
That is, the customer also needs to write API code to generate measurement data for calculated characteristics, is that right?

**Comment by czthiele** [comment]
Thats right, at the moment you would need to parse the calculation rule yourself and calculate the value.

We know that calculated characteristics are an important part of PiWeb and that currently an example which would answer your question is missing in the public documentation. We have plans to create such an example project, but it is not immediately planned.
I would notify you here if we have further information.

**Comment by Clint13** [comment]
Hello,
I have the same use case. We need to get calculated values via the API.

Do you have any timeline when you will add the functionality to the API?
